src/pages/iou/DistanceRequestPage.js

```diff
@@ -5,34 +5,62 @@ import * as IOU from '../../libs/actions/IOU';
 import ONYXKEYS from '../../ONYXKEYS';
 import DistanceRequest from '../../components/DistanceRequest';
 import reportPropTypes from '../reportPropTypes';
+import CONST from '../../CONST';
+import lodashGet from 'lodash/get';
 
 const propTypes = {
     /** The transactionID of this request */
-    transactionID: PropTypes.string,
+    iou: PropTypes.object,
 
     /** The report on which the request is initiated on */
     report: reportPropTypes,
+
+    /** Passed from the navigator */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            iouType: PropTypes.string,
+            reportID: PropTypes.string,
+        }),
+    }),
 };
 
 const defaultProps = {
-    transactionID: '',
+    iou: {
+        id: '',
+        amount: 0,
+        created: '',
+        merchant: '',
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
     report: {},
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
 };
 
 // This component is responsible for getting the transactionID from the IOU key, or creating the transaction if it doesn't exist yet, and then passing the transactionID.
 // You can't use Onyx props in the withOnyx mapping, so we need to set up and access the transactionID here, and then pass it down so that DistanceRequest can subscribe to the transaction.
-function DistanceRequestPage({report, transactionID}) {
+function DistanceRequestPage({iou, report, route}) {
+    const iouType = lodashGet(route, 'params.iouType', '');
+    const reportID = lodashGet(route, 'params.reportID', '');
+
     useEffect(() => {
-        if (transactionID) {
+        if (iou.transactionID) {
             return;
         }
         IOU.createEmptyTransaction();
-    }, [transactionID]);
+    }, [iou.transactionID]);
 
     return (
         <DistanceRequest
+            iou={iou}
+            iouType={iouType}
+            reportID={reportID}
             report={report}
```

Comment:
If `iouType` and `reportID` can be inferred from `iou` and `report`, then they shouldn't be passed as separate params.

Reply:
I don't think `iouType` doesn't exist on `iou` object.
<img width="350" alt="Screenshot 2023-08-24 at 3 29 41 PM" src="https://github.com/Expensify/App/assets/98560306/91de1527-21ff-4ff9-ad00-183dd9211316">